core/src/io/anuke/mindustry/world/Block.java

```diff
@@ -245,6 +245,38 @@ public void drawLayer(Tile tile){
     public void drawLayer2(Tile tile){
     }
 
+    public void drawBottleneck(Tile tile){
+        if(tile.entity == null) return;
+
+        switch(tile.block.category){
```

Comment:
What is this switch statement for? Are you only drawing bottlenecks for tiles that can have them? For those cases, I think it would suffice to check whether `!block.consumes.all().length > 0`.